sem/utils.py

```diff
@@ -330,3 +331,104 @@ def compute_sensitivity_analysis(
+            prefix_all will always be appended.
+
+            For example,
+            log_component = {
+                'component1' : 'info',
+                'component2' : 'level_debug|info'
+            }
+        ns3_log_components (list): A list containing all the valid log
+            components supported by ns-3.
+    """
+    if not log_component:
+        return None
+
+    ret_dict = {}
+    log_level_list = ['error',
+                      'warn',
+                      'debug',
+                      'info',
+                      'function',
+                      'logic']
+    converter = {
+        'error': ['error'],
+        'warn': ['warn'],
+        'debug': ['debug'],
+        'info': ['info'],
+        'function': ['function'],
+        'logic': ['logic'],
+        'all': ['error', 'warn', 'debug', 'info', 'function', 'logic'],
+        'level_error': ['error'],
+        'level_warn': ['error', 'warn'],
+        'level_debug': ['error', 'warn', 'debug'],
+        'level_info': ['error', 'warn', 'debug', 'info'],
+        'level_function': ['error', 'warn', 'debug', 'info', 'function'],
+        'level_logic': ['error', 'warn', 'debug', 'info', 'function', 'logic'],
+        'level_all': ['error', 'warn', 'debug', 'info', 'function', 'logic'],
+        '**': ['error', 'warn', 'debug', 'info', 'function', 'logic'],
+        '*': None,
+        'prefix_func': None,
+        'prefix_time': None,
+        'prefix_node': None,
+        'prefix_level': None,
+        'prefix_all': None
+    }
+    for component, levels in log_component.items():
+        log_level_complete = set()
+        if (ns3_log_components is not None) and not (component not in ns3_log_components or component != '*'):
+            raise ValueError(
+                "Log component '%s' is not a valid ns-3 log component. Valid log components: \n%ls" % (
+                    component,
+                    ns3_log_components
+                ))
+        for level in levels.split('|'):
+            if level not in converter:
+                raise ValueError("Log level '%s' for component '%s' is not valid"
+                                 % (level,
+                                    component))
+
+            # '*' represents level_all only if it occurs before the first '|'
+            if levels.split('|')[0] == '*':
+                log_level_complete.update(converter['all'])
+                continue
+
+            # Do not update the dictionary if prefixes are mentioned
+            if converter[level] is not None:
+                log_level_complete.update(converter[level])
+
+        # Update log_level_complete if entry for components exists
+        if component in ret_dict:
+            log_level_complete.update([log_level
+                                      for log_level in ret_dict[component].split('|')
+                                      if log_level not in log_level_complete])
```

Comment:
no need to make any check: log_level_complete is a set and already handles possible duplicates. am i missing something?
```suggestion
        if component in ret_dict:
            log_level_complete.update(ret_dict[component].split('|'))
```